drivers/spi/spi_esp32_spim.c

```diff
@@ -55,19 +58,43 @@ static inline void spi_esp32_complete(const struct device *dev,
 static int IRAM_ATTR spi_esp32_transfer(const struct device *dev)
 {
 	struct spi_esp32_data *data = dev->data;
+	const struct spi_esp32_config *cfg = dev->config;
 	struct spi_context *ctx = &data->ctx;
 	spi_hal_context_t *hal = &data->hal;
 	spi_hal_dev_config_t *hal_dev = &data->dev_config;
 	spi_hal_trans_config_t *hal_trans = &data->trans_config;
 	size_t chunk_len = spi_context_max_continuous_chunk(&data->ctx);
 	chunk_len = MIN(chunk_len, SOC_SPI_MAXIMUM_BUFFER_SIZE);
+	size_t bit_len = chunk_len << 3;
+	uint8_t *rx_temp = NULL;
+	uint8_t *tx_temp = NULL;
+
+	if (cfg->dma_enabled) {
+		bit_len = (!chunk_len) ? (chunk_len + 1) << 3 : bit_len;
```

Comment:
Not so trivial, I would like to see some comment here

Reply:
Simplified the expression and added the comment :)